doc/modules/partial_dependence.rst

```diff
@@ -163,16 +170,87 @@ support it, and 'brute' is used for the rest.
-   function.
+Individual conditional expectation (ICE) plot
+=============================================
+
+Similar to a PDP, an individual conditional expectation (ICE) plot
+shows the dependence between the target function and a target feature.
+However, unlike a PDP, which shows the average effect of the target
+feature, an ICE plot visualizes the dependence of the prediction on a
+feature for each instance separately with one line per instance.
+Due to the limits of human perception, only one target feature is
+supported for ICE plots.
+
+The figures below show four ICE plots for the California housing dataset,
+with a :class:`GradientBoostingRegressor
+<sklearn.ensemble.GradientBoostingRegressor>`. The second figure plots
+the corresponding PD line overlaid on ICE lines.
+
+.. figure:: ../auto_examples/inspection/images/sphx_glr_plot_partial_dependence_004.png
+   :target: ../auto_examples/inspection/plot_partial_dependence.html
+   :align: center
+   :scale: 70
+
+.. figure:: ../auto_examples/inspection/images/sphx_glr_plot_partial_dependence_005.png
+   :target: ../auto_examples/inspection/plot_partial_dependence.html
+   :align: center
+   :scale: 70
+
+While the PDPs are good at showing the average effect of the target features,
+they can obscure a heterogeneous relationship created by interactions.
+When interactions are present the ICE plot will provide much more insight.
+For example, we could observe a linear relationship between the median income
+and the house price in the PD line. However, the ICE lines show that there
+are some exceptions, where the house price remains constant in some ranges of
+the median income.
+
+Similar to PDPs, the target features are assumed to be independent from the
+complement features.
+
+The :mod:`sklearn.inspection` module's :func:`plot_partial_dependence`
+convenience function can be used to create ICE plots by setting
+``individual=True``. In the below example we show how to create a grid of
+ICE plots:
+
+    >>> from sklearn.datasets import make_hastie_10_2
+    >>> from sklearn.ensemble import GradientBoostingClassifier
+    >>> from sklearn.inspection import plot_partial_dependence
+
+    >>> X, y = make_hastie_10_2(random_state=0)
+    >>> clf = GradientBoostingClassifier(n_estimators=100, learning_rate=1.0,
+    ...     max_depth=1, random_state=0).fit(X, y)
+    >>> features = [0, 1]
+    >>> plot_partial_dependence(clf, X, features,
+    ...     individual=True) #doctest: +SKIP
+
+You can access the newly created figure and Axes objects using ``plt.gcf()``
+and ``plt.gca()``.
+
+In ICE plots it might not be easy to see the average effect of the target
+variable. Hence, it is recommended to use ICE plots alongside partial
+dependency plots. They can be plotted together with ``individual='both'``.
+
+    >>> plot_partial_dependence(clf, X, features,
+    ...     individual='both') #doctest: +SKIP
+
+Mathematical Definition
```

Comment:
I feel this section a bit alone. Would it makes sense to have 

1. PDP
2. ICE
3. Mathematical defintion for PDP and ICE

@NicolasHug any thoughts?

Reply:
I have moved 'Mathematical definitions' and 'Computation method' sections to the bottom in https://github.com/scikit-learn/scikit-learn/pull/16619/commits/91258cae84afcd510b17cc1338d47411785b5806

Reply:
> Also we should update the paragraph on the method to explain that computing the individual conditional expectations is only possible when using the "brute" method. The faster "recursion" method is can only compute the average ICE values.
https://github.com/scikit-learn/scikit-learn/pull/16619/commits/a8a77f4400a7de1461498bf8ce94a620813e6715